flair/models/sequence_tagger_model.py

```diff
@@ -252,41 +255,57 @@ def forward_labels_and_loss(self, sentences: Union[List[Sentence], Sentence]) ->
             tags = self._obtain_labels(feature, lengths)
             return tags, loss
 
-    def predict(self, sentences: Union[List[Sentence], Sentence], mini_batch_size=32) -> List[Sentence]:
+    def predict(self, sentences: Union[List[Sentence], Sentence],
+                mini_batch_size=32, verbose=False) -> List[Sentence]:
         with torch.no_grad():
-            if type(sentences) is Sentence:
+            if isinstance(sentences, Sentence):
                 sentences = [sentences]
 
             filtered_sentences = self._filter_empty_sentences(sentences)
 
             # remove previous embeddings
             clear_embeddings(filtered_sentences, also_clear_word_embeddings=True)
 
+            # revere sort all sequences by their length
+            filtered_sentences.sort(key=lambda x: len(x), reverse=True)
+
             # make mini-batches
             batches = [filtered_sentences[x:x + mini_batch_size] for x in
                        range(0, len(filtered_sentences), mini_batch_size)]
 
-            for batch in batches:
+            # progress bar for verbosity
+            if verbose:
+                batches = tqdm(batches)
+
+            for i, batch in enumerate(batches):
+
+                if verbose:
+                    batches.set_description(f'Inferencing on batch {i}')
+
                 tags, _ = self.forward_labels_and_loss(batch)
 
                 for (sentence, sent_tags) in zip(batch, tags):
                     for (token, tag) in zip(sentence.tokens, sent_tags):
                         token: Token = token
                         token.add_tag_label(self.tag_type, tag)
 
+                # clearing token embeddings to save memory
+                clear_embeddings(batch, also_clear_word_embeddings=True)
+
             return sentences
 
     def forward(self, sentences: List[Sentence]):
         self.zero_grad()
 
         self.embeddings.embed(sentences)
 
-        # first, sort sentences by number of tokens
-        sentences.sort(key=lambda x: len(x), reverse=True)
-        longest_token_sequence_in_batch: int = len(sentences[0])
+        # if grad is enabled, sort sentences by number of tokens
+        if torch.is_grad_enabled():
+            sentences.sort(key=lambda x: len(x), reverse=True)
 
         lengths: List[int] = [len(sentence.tokens) for sentence in sentences]
         tag_list: List = []
+        longest_token_sequence_in_batch: int = len(sentences[0])
```

Comment:
during eval phase of training, grad is not enabled, causing sentences to not be sorted. the longest sentence is then not always in the first position (sentences[0]). 

this raises an exception during training (eval phase). 

to reproduce error: 

```python
corpus: TaggedCorpus = NLPTaskDataFetcher.load_corpus(NLPTask.UD_ENGLISH).downsample(0.01)

tag_dictionary = corpus.make_tag_dictionary(tag_type='pos')

tagger: SequenceTagger = SequenceTagger(hidden_size=256,
                                        embeddings=WordEmbeddings('glove'),
                                        tag_dictionary=tag_dictionary,
                                        tag_type='pos')

trainer: ModelTrainer = ModelTrainer(tagger, corpus)

trainer.train('resources/taggers/example-ner')

```

Reply:
Ah yes, I thought the predict method would be used in the training phase (which now sorts the sequences). I'll see what can be done here. Thanks!

Reply:
I made the fix. But, how is that the test's passed both locally and on travis?

Reply:
Yeah I was wondering about this as well, the test should not have passed. The problem was that the error occurs only if the first sentence is not longest sentence in a mini-batch, which in the tiny datasets we use for integration tests apparently never happens. We need to add an additional test for this.